fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/jbatch/load/ChunkWriter.java

```diff
@@ -248,4 +261,68 @@ public void writeItems(List<java.lang.Object> arg0) throws Exception {
             throw e;
         }
     }
+
+    /**
+     * conditional update checks to see if our cache contains the key, if not reads from the db, and calculates the cache.
+     * The cache is saved within the context of this particular execution, and then destroyed.
+     *
+     * @implNote considered using a shared cache, a few things with that to consider:
+     * 1 - the shared cache would have to be updated at the end of a transaction (we don't control it).
+     * 2 - we would have to use a transaction sync registry to control the synchronization of the cache.
+     * 3 - Instead, we're doing a read then update.
+     *
+     * @param chunkData the transient user data used increment the number of skips
+     * @param skip should skip the resource if it matches
+     * @param localCache map containing the key-saltHash
+     * @param persistence used to facilitate the calls to the underlying db
+     * @param context used in db calls
+     * @param logicalId the logical id of the FHIR resource (e.g. 1-2-3-4)
+     * @param resource the FHIR Resource
+     * @return outcomes including information or warnings
+     * @throws FHIRPersistenceException
+     */
+    public OperationOutcome conditionalFingerprintUpdate(ImportTransientUserData chunkData, boolean skip, Map<String, SaltHash> localCache, FHIRPersistence persistence, FHIRPersistenceContext context, String logicalId, Resource resource) throws FHIRPersistenceException {
```

Comment:
We should probably try to avoid the extra read + fingerprinting in cases where we're not even going to skip the updates when the fingerprints match.

Reply:
Good catch, I had intended to add a skip on 290, when I add it it'll skip the extra read.

Reply:
This addresses my concern for the extra read, but its still computing and caching the resource fingerprint, even though that isn't used.  Not sure how much that costs, but I think we should skip that as well.

Reply:
I added a wrap to the skip.  thanks for the catch